### ISSUE TITLE
Example `ticker_updates.py` fails on Python 3

### ISSUE THREAD

**Issue by ferrerluis** [slazarov/python-bittrex-websocket#39]
When I download and run this example, it first complains that `ticker_updates_container` is not an attribute of `ws`. I assume this means `on_open()` is not running on time before the while loop starts running. Therefore, I overrode the `__init__` for `MySocket` as follows:

```python
def __init__(self):
    super().__init__()
    self.ticker_updates_container = {}
```

Even then, I get this error later:

```shell
$ python ticker_updates.py
2018-02-20 18:21:36 - bittrex_websocket.websocket_client - INFO - [Connection][009cd9f08d954f7d91d5d59f15c8b583]:Trying to establish connection to Bittrex through https://socket-stage.bittrex.com/signalr.
2018-02-20 18:21:41 - bittrex_websocket.websocket_client - ERROR - [Connection][009cd9f08d954f7d91d5d59f15c8b583]:Timeout for url https://socket-stage.bittrex.com/signalr. Please check your internet connection is on.
2018-02-20 18:21:41 - bittrex_websocket.websocket_client - INFO - [Connection][009cd9f08d954f7d91d5d59f15c8b583]:Trying to establish connection to Bittrex through https://socket.bittrex.com/signalr.
2018-02-20 18:21:47 - bittrex_websocket.websocket_client - ERROR - [Connection][009cd9f08d954f7d91d5d59f15c8b583]:Timeout for url https://socket.bittrex.com/signalr. Please check your internet connection is on.
2018-02-20 18:21:47 - bittrex_websocket.websocket_client - ERROR - [Connection][009cd9f08d954f7d91d5d59f15c8b583]:Failed to establish connection through supplied URLS. Leaving to watchdog...
2018-02-20 18:21:56 - bittrex_websocket.websocket_client - ERROR - Failed to subscribe [TickerUpdate][['BTC-ETH', 'BTC-NEO', 'BTC-ZEC', 'ETH-NEO', 'ETH-ZEC']] from connection 009cd9f08d954f7d91d5d59f15c8b583 after 20 seconds. The connection is probably down.
```

I know the Node library [node.bittrex.api](https://github.com/dparlevliet/node.bittrex.api) uses the WebSocket URL `wss://socket.bittrex.com/signalr` and that library works for me.

Could it be an issue with the URL? If so, can I change the URL through options without changing `BittrexSocket`?

**Comment by ferrerluis** [comment]
UPDATE:
Tried running it with the Docker container provided in the repo and it still can't establish a connection

**Comment by AlnsV** [comment]
Same thing is happening to me, i'm subscribing to trades of a bunch of pairs but no connection, i also tried different versions (0.0.5.1, 0.0.6.2.2 and 0.0.6.3)

**Comment by slazarov** [comment]
I will test later and will provide info.

**Comment by slazarov** [comment]
I have no issues, connection is safely established and working. It might have been something temporary. You can change the socket addresses from `_init_connection` but I have checked the available ones and they haven't changed.

Make sure you are:
- on the latest version of the socket = 0.0.6.3
- cfscrape = 1.9.4
- node.js is installed

**Comment by arnaldojvg** [comment]
i did those updates and have the error messages still...

**Comment by arnaldojvg** [comment]
$ pip show cfscrape
Name: cfscrape
Version: 1.9.4
Summary: A simple Python module to bypass Cloudflare's anti-bot page. See https://github.com/Anorov/cloudflare-scrape for more information.
Home-page: https://github.com/Anorov/cloudflare-scrape
Author: Anorov
Author-email: anorov.vorona@gmail.com
License: UNKNOWN
Location: /usr/local/lib/python3.5/dist-packages
Requires: requests, PyExecJS
$ show bittrex_websocket
Name: bittrex-websocket
Version: 0.0.6.3
Summary: The unofficial Python websocket client for the Bittrex Cryptocurrency Exchange
Home-page: https://github.com/slazarov/python-bittrex-websocket
Author: Stanislav Lazarov
Author-email: s.a.lazarov@gmail.com
License: MIT
Location: /usr/local/lib/python3.5/dist-packages
Requires: cfscrape, signalr-client, requests, Events, websocket-client, gevent
$ nodejs -v
v4.2.6

and still have the issue

2018-02-21 21:56:05 - bittrex_websocket.websocket_client - ERROR - [Connection][d8c9309ef10947bbbd07a7f03a7d00a2]:Timeout for url https://socket.bittrex.com/signalr. Please check your internet connection is on.
2018-02-21 21:56:05 - bittrex_websocket.websocket_client - INFO - [Connection][d8c9309ef10947bbbd07a7f03a7d00a2]:Trying to establish connection to Bittrex through wss://socket.bittrex.com/signalr.
2018-02-21 21:56:05 - bittrex_websocket.websocket_client - ERROR - [Connection][d8c9309ef10947bbbd07a7f03a7d00a2]:Timeout for url wss://socket.bittrex.com/signalr. Please check your internet connection is on.
2018-02-21 21:56:05 - bittrex_websocket.websocket_client - ERROR - [Connection][d8c9309ef10947bbbd07a7f03a7d00a2]:Failed to establish connection through supplied URLS. Leaving to watchdog...
2018-02-21 21:56:15 - bittrex_websocket.websocket_client - ERROR - Failed to subscribe [Trades][['BTC-LTC', 'BTC-DOGE', 'BTC-VTC', 'BTC-PPC', 'BTC-FTC', 'BTC-RDD', 'BTC-NXT', 'BTC-DASH', 'BTC-POT', 'BTC-BLK', 'BTC-EMC2', 'BTC-XMY', 'BTC-AUR', 'BTC-EFL', 'BTC-GLD', 'BTC-SLR', 'BTC-PTC', 'BTC-GRS', 'BTC-NLG', 'BTC-RBY']] from connection d8c9309ef10947bbbd07a7f03a7d00a2 after 20 seconds. The connection is probably down.
2018-02-21 21:56:15 - bittrex_websocket.websocket_client - INFO - [Connection][39d6e6ac36364e85b8a1a16bb319f29d]:Trying to establish connection to Bittrex through https://socket-stage.bittrex.com/signalr.

**Comment by slazarov** [comment]
Is node.js installed on your computer?

**Comment by arnaldojvg** [comment]
yes i do 
$ nodejs -v
v4.2.6

**Comment by emestee** [comment]
Do you have any proxy-related environment variables? websocket and signalr mishandle it a bit, if you have any sort of proxying enabled this is what would happen

**Comment by AlnsV** [comment]
We have tried in different computers and it didn't work in any of them. But i would take a look to the environment variables you sugest

**Comment by slazarov** [comment]
Can you try from different network?
  

  
  

  
  
>

**Comment by arnaldojvg** [comment]
We have tried in my home network, the server network, his home network and a co-workers home network, neither works.

**Comment by slazarov** [comment]
Looks like #27 to me
https://github.com/slazarov/python-bittrex-websocket/issues/27

**Comment by slazarov** [comment]
Try running in a venv and make sure you install request with security parameter as per the readme.

**Comment by slazarov** [comment]
The error you are getting is from `from urllib3.contrib.pyopenssl import SocketError`

**Comment by AlnsV** [comment]
Thank you. I have not found the way to fix it tho

**Comment by slazarov** [comment]
Did you try a python3 venv?

**Comment by AlnsV** [comment]
Yes i did.

**Comment by slazarov** [comment]
Can you retry the docker config from the new PR?

**Comment by slazarov** [comment]
Closing due to inactivity.
